socrata/socrata_upload.py

```diff
@@ -87,19 +95,16 @@ def build_query(
     # Retrieve column names and types from Athena
     columns = cursor.execute("show columns from " + athena_asset).as_pandas()
 
-    # Limit pull to columns present in open data asset
-    asset_url = (
-        "https://datacatalog.cookcountyil.gov/resource/"
-        + asset_id
-        + ".json?$limit=1"
-    )
-
+    # Limit pull to columns present in open data asset - shouldn't change anything, but prevents failure if columns have become misaligned.
     asset_columns = (
         requests.get(
-            asset_url, headers={"X-App-Token": os.getenv("SOCRATA_APP_TOKEN")}
+            f"https://datacatalog.cookcountyil.gov/resource/{asset_id}"
         )
-        .json()[0]
-        .keys()
+        .headers["X-SODA2-Fields"]
+        .replace('"', "")
+        .strip("[")
+        .strip("]")
+        .split(",")
```

Comment:
I was very naively pulling asset column names before. If the first row of an asset had `NULL` values for a column, we wouldn't get back that column header. This ensures we get all column headers.